lib/pbench/test/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""Common test support functions."""
+import json
+
+from filelock import FileLock
+from pathlib import Path
+from typing import Callable
+
+
+def on_disk_config(
+    tmp_path_factory, prefix: str, setup_func: Callable[[Path], Path]
+) -> dict:
+    """Base on-disk configuration recorder.  It returns a dictionary with two
+    items, "tmp" and "cfg_dir", two Path objects for the temporary directory
+    in which the configuration directory is created.
+
+    The first argument is the temporary directory creator factory to be used.
+
+    The second argument is a prefix to use for the JSON marker file in which
+    will be store the dictionary containing the temporary directory and the
+    configuration directory created by the setup_func.
+
+    The third argument is the setup function itself.
```

Comment:
The reason I kept it that way is that the parameter class is not exposed anywhere that I could find to import.  So keeping it the same name as the fixture seemed the best course of action.

I can change the comment use the parameter names.

Reply:
Changing the comment would be good, regardless.

I don't see how the lack of an import drives the name of the parameter.  Given that this is Python, we don't actually have to know the type of the parameters until call-time (and, even then, we can be flexible...).  Keeping it the same as the fixture seems likely to create confusion:  that is, receiving the value of a fixture is different from referencing one -- I think it would be good to be clear that this function is not referencing the fixture (i.e., it is not causing the fixture to be included/invoked), rather it is making use of the value that the fixture provides to this function's caller.

(And, yeah, I'm willing to take a pass on the type hint for it...although, I s'pose we could use `Any`, but that's probably no better than just omitting it.)

Reply:
First, a non-test function physically can't reference a fixture.  It is a parameter argument, and so it is always the name of the first parameter.  I'd rather keep the argument to be the same name as the fixture for two reasons:

 1. Naming it something else is likely to cause confusion because we can't type hint it, and so unless you really understand the nature of the callers and their fixture, it won't be clear what to pass here
 2. I don't know what to name it that would help make it any clearer that we really want to use a `TempPathFactory` class only available from a fixture of the name `tmp_path_factory`

Reply:
> a non-test function physically can't reference a fixture

Right(*)...which is a big part of why it is so confusing to see one look like it's doing exactly that.

> Naming it something else is likely to cause confusion because we can't type hint it

Aside from the fact that we _can_ apply a hint to it (if nothing else, I believe that we can put the hint in double quotes to escape definition problems, which means we can label it anything that we think will help the reader)...the name only has to be meaningful within the context of the function (once we understand that it's not a fixture...).  So, I think we can find a spelling which is expressive without being confusing.

> I don't know what to name it

We could call it `path_factory`, since, in this context, I don't think we care whether it is temporary or not....

Anyway, I think we have given this issue due consideration, and I think we have bigger fish to fry, so I'm going to mark this resolved, now.


---------
(*)Ignoring the fact that other fixtures are also non-test functions which _can_ reference other fixtures.